test/parallel/xyk-pallet.API.poolVestedToken.test.ts

```diff
@@ -0,0 +1,360 @@
+    );
+    expect(getEventResultFromMangataTx(burnLiudityResult).data).toContain(
+      "NotEnoughAssets"
+    );
+
+    const transferAssetResult = await transferAsset(
+      testUser1.keyRingPair,
+      liquidityID,
+      sudo.keyRingPair.address.toString(),
+      new BN(150000)
+    );
+    expect(getEventResultFromMangataTx(transferAssetResult).state).toEqual(
+      ExtrinsicResult.ExtrinsicFailed
+    );
+    expect(getEventResultFromMangataTx(transferAssetResult).data).toContain(
+      "LiquidityRestrictions"
+    );
+  });
+
+  test("xyk-pallet- check that unlocking tokens are free to use when the relock time is done", async () => {
+    const api = getApi();
+
+    await createPoolAndVestingToken(true, toBN("1", 20), new BN(100));
+
+    const UserBalanceBeforeAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+
+    expect(
+      UserBalanceBeforeAmount.free.sub(UserBalanceBeforeAmount.frozen)
+    ).bnEqual(new BN(0));
+
+    const unlockSomeVestedToken = await unlockVestedToken(
+      testUser1,
+      liquidityID
+    );
+    expect(getEventResultFromMangataTx(unlockSomeVestedToken).state).toEqual(
+      ExtrinsicResult.ExtrinsicSuccess
+    );
+
+    const UserBalanceAfterUnlockingAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+    expect(
+      UserBalanceAfterUnlockingAmount.free.sub(
+        UserBalanceAfterUnlockingAmount.frozen
+      )
+    ).bnGt(new BN(0));
+
+    //@ts-ignore
+    const maxInstantBurnAmount = await api.rpc.xyk.get_max_instant_burn_amount(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+
+    const burnUnlockedToken = await burnLiquidity(
+      testUser1.keyRingPair,
+      MGA_ASSET_ID,
+      createdToken,
+      maxInstantBurnAmount
+    );
+    expect(getEventResultFromMangataTx(burnUnlockedToken).state).toEqual(
+      ExtrinsicResult.ExtrinsicSuccess
+    );
+
+    const UserBalanceAfterBurningAmount = await api.query.tokens.accounts(
+      testUser1.keyRingPair.address,
+      liquidityID
+    );
+    expect(
+      UserBalanceAfterBurningAmount.free.sub(
+        UserBalanceAfterBurningAmount.frozen
+      )
+    ).bnEqual(new BN(0));
+
+    expect(
+      UserBalanceBeforeAmount.frozen.sub(UserBalanceAfterBurningAmount.frozen)
+    ).bnGt(new BN(0));
+  });
+
+  test("xyk-pallet- check that all unlocking vesting tokens can be burned", async () => {
+    const api = getApi();
+    const poolLockedValue = toBN("1", 20);
+    const poolBlockValue = toBN("2", 19);
+
+    const VestingTokenFunction = await createPoolAndVestingToken(
+      true,
+      poolLockedValue,
+      poolBlockValue
+    );
+
+    const vestingStartBlockNumber = new BN(
+      VestingTokenFunction.vestingStartBlockNumber
+    );
+
+    const vestingFinishBlockNumber = new BN(
+      vestingStartBlockNumber.add(
+        poolLockedValue.sub(poolBlockValue).div(poolBlockValue)
```

Comment:
For example:

1000 vested 
Each block 100
-> after   10 blocks all of them will be unblock

why? 

( 1000 - 100 )  / amountUnlockEachBlock 

Should not be?
Formula:
vested / amountUnlockEachBlock -> number of blocks when scheduling is finished.
wait until startBlockNumber + `number of blocks when scheduling is finished` +1 (just in case)

Reply:
Yeah, my fault. Can't remember why I decided to subscribe amount first. Change it.